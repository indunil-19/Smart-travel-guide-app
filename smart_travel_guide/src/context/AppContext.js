import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}