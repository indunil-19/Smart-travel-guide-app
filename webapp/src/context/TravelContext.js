import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 