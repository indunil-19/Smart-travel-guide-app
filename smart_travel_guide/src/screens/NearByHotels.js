import React, { useEffect, useState, useContext } from "react";
import {
  Text,
  TouchableOpacity,
  Image,
  FlatList,
  View,
  Pressable,
  StyleSheet,
} from "react-native";
import {
  ActivityIndicator,
  Title,
  Button,
  FAB,
  Portal,
  Provider,
} from "react-native-paper";
import { getNearByHotels } from "../services/ViewPOIServices";
import { theme } from "../core/theme";
import { AppContext } from "../context/AppContext";
import { LocationInfoCard } from "../components/LocationInfoCard";

export const NearByHotels = ({ navigation }) => {
  const [data, setData] = useState([]);
  const { state, dispatch } = useContext(AppContext);

  useEffect(() => {
    getNearByHotels({ lat: 6.927079, lng: 79.85775 }).then((r) => {
      setData(r);
    });
  }, [state.accomodation_location]);
  return (
    <View
      style={{
        flex: 1,
        flexDirection: "column",
        justifyContent: "flex-end",
      }}
    >
      <Pressable
        style={[StyleSheet.absoluteFill]}
        onPress={navigation.goBack}
      />
      <View
        style={{
          height: "100%",
          width: "100%",
          backgroundColor: "#fff",
          justifyContent: "space-evenly",
          alignItems: "stretch",
          paddingHorizontal: 10,
          paddingTop: 10,
          height: "70%",
          borderTopEndRadius: 20,
          borderTopStartRadius: 20,
        }}
      >
        {!data && (
          <>
            <Image
              style={{ width: 150, height: 150 }}
              source={require("../assets/img/traveller.png")}
            />
            <ActivityIndicator animating={true} size={80} theme={theme} />
          </>
        )}
        {data && (
          <FlatList
            data={data}
            keyExtractor={(item, index) => item.place_id + index}
            ListHeaderComponent={
              <View style={{ alignItems: "center", justifyContent: "center" }}>
                <Title
                  style={{ color: theme.colors.primary, alignItems: "center" }}
                >
                  LOOK UP HOTELS
                </Title>
              </View>
            }
            renderItem={({ item, index }) => (
              <TouchableOpacity
                onPress={() => {
                  navigation.navigate("Location Detail", {
                    id: item.place_id,
                    name: item.name,
                  });
                }}
              >
                <LocationInfoCard
                  location={item}
                  photo={item.photos ? item.photos[0].photo_reference : ""}
                />
              </TouchableOpacity>
            )}
          />
        )}
      </View>
    </View>
  );
};
