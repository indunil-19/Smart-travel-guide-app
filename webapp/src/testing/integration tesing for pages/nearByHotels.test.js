import React from "react"
import {screen, cleanup,waitFor} from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import {TravelProvider} from "../../context/TravelContext"
import "@testing-library/jest-dom/extend-expect";
import NearByHotels from "../../screens/TravelPlanApp/nearbyhotels"

describe('view poi page test', () => {
    
    
    afterEach(cleanup)

    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><NearByHotels/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><NearByHotels /></TravelProvider></MemoryRouter>,div)

        
    })

    it("should render nearby places if no errorr",()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><NearByHotels /></TravelProvider></MemoryRouter>,div)

    })

    
})
