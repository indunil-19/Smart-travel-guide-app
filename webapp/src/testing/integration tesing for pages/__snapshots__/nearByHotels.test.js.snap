// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view poi page test should match snapshot 1`] = `
<div
  className="css-dvxtzn"
/>
`;
